jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Result.java

```diff
@@ -52,18 +51,20 @@
 	@Getter private final String className;
 	@Getter private final String objDomain;
 	@Getter private final String typeName;
-	@Nonnull @Getter private final ImmutableMap<String, Object> values;
+	@Getter private final ImmutableList<String> valuePath;
```

Comment:
I almost think that valuePath should be its own type, completely hiding the implementation. That being said, you have read all that messy code more recently than I have, so feel free to have a different opinion!

Reply:
What do you imagine in this specific type?

Reply:
Actually, I'm imagining something more generic than just `valuePath` (I was reading all this a bit too fast). If I understand correctly, `valuePath` is the path specific to the value, in the context of the specific bean. I was imagining a more generic abstraction, which would care about the full path (or name, or context, or coordinate, not sure about the name) of any value. This more generic object would abstract both the `valuePath` and what is inferred from the `className`, `objName`, `typeName`, ... But this probably belong to another change.

Reply:
At the moment, valuePath contains the path from the MBean attribute to the simple (numeric) value:
* Empty when the attribute is already a simple value (java.lang:type=Threading ThreadCount [])
* Key when the attribute is Map or a CompositeData (java.lang:type=Memory HeapMemoryUsage [used])
* Numerical index when the attribute is Array or Collection 
* Row Key when the attribute is TabularData (java.lang:type=Runtime SystemProperties [java.version, value] or [java.version, key])

Reply:
yep, that make sense as a first step!